src/front/glsl/lex.rs

```diff
@@ -93,13 +93,12 @@ impl<'a> Lexer<'a> {
                 }
             }
             '0'..='9' => {
-                let (number, _, pos) =
+                let (number, remainder, pos) =
                     consume_any(input, |c| (('0'..='9').contains(&c) || c == '.'));
                 if number.find('.').is_some() {
-                    if (
-                        chars.next().map(|c| c.to_lowercase().next().unwrap()),
-                        chars.next().map(|c| c.to_lowercase().next().unwrap()),
-                    ) == (Some('l'), Some('f'))
+                    if remainder
+                        .get(..2)
+                        .map_or(false, |c| &c.to_lowercase() == "lf")
```

Comment:
nit: seems unfortunate that we are calling `to_lowercase` multiple times, since it involves a string allocation on the heap. Ideally, we should not call it even once, since we are just comparing slices. I.e. could this be something like
```rust
let mut remainder_chars = remainder.chars();
let first = remainder_chars.next();
let second = remainder_chars.next();
match (first, second) {
  (Some(a), Some(b)) if a.to_lowercase() == 'l' && b.to_lowercase == 'f' => {..}
  ...
}
```